### DISCUSSION TITLE
Can Figma manifest.json ui field be a JS file?

### DISCUSSION THREAD

**Discussion by Helium-Z** [yuanqing/create-figma-plugin#230] [General]
Hi, I have tried your wonderful tool to build my Figma plugin. I'm curious whether the `ui` field of Figma `manifest.json` can be a JS file. I found nothing about this on Figma plugin documents. How did you know you could set up a JS entry for it?

**Comment by Helium-Z** [comment]
I notified that build-task build `build/ui.js` from [these contents](https://github.com/yuanqing/create-figma-plugin/blob/main/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts#L162). So `ui.js` will be executed after `main.js` in Figma plugin?

**Comment by yuanqing** [comment]
Whatever file that’s pointed to in the `ui` field of the `manifest.json` will be read as a string and be accessible via the `__html__` global variable. See the [official docs](https://www.figma.com/plugin-docs/creating-ui/)
